core/src/main/scala/org/apache/spark/util/DependencyUtils.scala

```diff
@@ -316,7 +316,7 @@ private[spark] object DependencyUtils extends Logging {
       case _ =>
         val fs = FileSystem.get(uri, hadoopConf)
         Option(fs.globStatus(new Path(uri))).map { status =>
-          status.filter(_.isFile).map(_.getPath.toUri.toString)
+          status.map(_.getPath.toUri.toString)
```

Comment:
Does it really work? It's added later via `SparkContext.addFile` in the constructor of `SparkContext` which disables `recursive` by default.

Reply:
Also it affects all other functionalities such as jars. I don't think it makes much sense to allow directories for them.

Reply:
yes, it worked, and jars are using addJar function